sig-platform-dev/README.md

```diff
@@ -0,0 +1,83 @@
+<!---
+This is an autogenerated file!
+
+Please do not edit this file directly, but instead make changes to the
+sigs.yaml file in the project root.
+
+To understand how this file is generated, see https://git.k8s.io/community/generator/README.md
+-->
+# Platform Dev Special Interest Group
+
+SIG Platform Dev (SIG-PD) is a cross-cutting SIG whose mission is to design, build, and maintain a cohesive, first-party Platform Development Kit (PDK) for developers who want to extend the Kubernetes platform by adding new APIs or enhancing existing APIs, as well as to advocate on behalf of such developers by driving feedback into other SIGs.
+
+## Meetings
+* Regular SIG Meeting: [Tuesdays at 10:00 PT (Pacific Time)]() (biweekly). [Convert to your timezone](http://www.thetimezoneconverter.com/?t=10:00&tz=PT%20%28Pacific%20Time%29).
+
+## Leadership
+
+### Chairs
+The Chairs of the SIG run operations and processes governing the SIG.
+
+* Jimmy Zelinskie (**[@jzelinskie](https://github.com/jzelinskie)**), Red Hat
+* Anthony Yeh (**[@enisoc](https://github.com/enisoc)**), Google
+
+### Technical Leads
+The Technical Leads of the SIG establish new subprojects, decommission existing
+subprojects, and resolve cross-subproject technical issues and decisions.
+
+* Jimmy Zelinskie (**[@jzelinskie](https://github.com/jzelinskie)**), Red Hat
+* Anthony Yeh (**[@enisoc](https://github.com/enisoc)**), Google
+* Phillip Wittrock (**[@pwittrock](https://github.com/pwittrock)**), Google
+
+## Contact
+* [Slack](https://kubernetes.slack.com/messages/sig-platform-dev)
+* [Mailing list](https://groups.google.com/forum/#!forum/kubernetes-sig-platform-dev)
+* [Open Community Issues/PRs](https://github.com/kubernetes/community/labels/sig%2Fplatform-dev)
+
+## Subprojects
+
+The following subprojects are owned by sig-platform-dev:
+- **pdk-roadmap**
+  - Owners:
+    - https://raw.githubusercontent.com/kubernetes-sigs/pdk-roadmap/master/OWNERS
+- **sample-apiserver**
+  - Owners:
+    - https://raw.githubusercontent.com/kubernetes/sample-apiserver/master/OWNERS
+    - https://raw.githubusercontent.com/kubernetes/kubernetes/staging/src/k8s.io/sample-apiserver/master/OWNERS
+- **sample-controller**
+  - Owners:
+    - https://raw.githubusercontent.com/kubernetes/sample-controller/master/OWNERS
+    - https://raw.githubusercontent.com/kubernetes/kubernetes/staging/src/k8s.io/sample-controller/master/OWNERS
+- **gengo**
+  - Owners:
+    - https://raw.githubusercontent.com/kubernetes/gengo/master/OWNERS
+- **code-generator**
```

Comment:
what is the rational to have gengo+code-generator here? This looks like foundational tech that is deeply linked to API machinery.

Reply:
You're probably right since you know these tools much better than I do. For reference, my initial rationale was to pull out things that "generate concrete code that calls into abstract SIG-AM libraries", as opposed to the abstract libraries themselves that SIG-AM writes by hand.

For example: SIG-AM defines abstract interfaces with methods like `DeepCopy()`, `DeepCopyInto()`, `DeepCopyObject()`, etc. They don't really care how you choose to implement those interfaces, just that you conform to them.

It makes sense to me for SIG-PD to own a tool like deep-copy generator that gives you the option of implementing those interfaces through code generation. You could also choose to implement `DeepCopy()` based on reflection, or hand-write all your copies, or use a different generator.

Similarly, I envision SIG-AM owning client-go libraries like [rest](https://godoc.org/k8s.io/client-go/rest) and [cache](https://godoc.org/k8s.io/client-go/tools/cache), which implement abstract behavior for talking to API servers.

It makes sense to me for SIG-PD to own a tool like client-gen that gives you the option of wrapping these abstract libraries in generated, statically-typed code. Neither client-go nor the API server cares if you use this generator though. They are perfectly happy if you use the [dynamic client](https://godoc.org/k8s.io/client-go/dynamic), or manually instantiate a RESTClient and your own Reflector, Indexer, and Informer as needed, rather than generate a statically-typed factory for these things.

Reply:
I am not sure that I agree these tools should be subprojects of the SIG.  The test-framework will still be owned by SIG-testing, the docs generation will still be owned by SIG-docs.  The goal of the SIG is to integrate components owned by other SIGs into user friendly toolkit.  IMHO most of components it is integrating shouldn't be owned by the SIG unless that is good cause.

Reply:
Integrating tools owned by multiple SIGs is a good way of putting it. For example, kubebuilder integrates both client generation (clients are in scope for SIG-AM) and controller generation (controllers are out of scope for SIG-AM).

I'm not opposed to leaving these in SIG-AM if there is consensus that it makes sense.

Reply:
I guess one question is how tightly coupled are these to the apimachinery libraries.  IIRC they are relatively tightly coupled.  If that is the case, I think the ownership should stay together.

Reply:
Their output is tightly coupled with api machinery, i.e. the output is useless (maybe with the exception of deepcopy) outside of api machinery.

I also like the term "Integrating tools" as a rule of thumb for Sig-PD.